lib/elixir_search_extractor_worker/csv_keywords_processing_worker.ex

```diff
@@ -0,0 +1,20 @@
+defmodule ElixirSearchExtractor.ElixirSearchExtractorWorker.CsvKeywordsProcessingWorker do
+  use Oban.Worker
+  alias ElixirSearchExtractor.FileUpload.CsvParser
+  alias ElixirSearchExtractor.FileUpload.FileUploads
+  alias ElixirSearchExtractor.FileUpload.Schemas.KeywordFile
+  alias ElixirSearchExtractor.Repo
+  alias ElixirSearchExtractor.SearchKeyword.SearchKeywords
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"file_record_id" => record_id}}) do
+    file_record = Repo.get_by!(KeywordFile, %{id: record_id})
+
+    with {:ok, keyword_list} <- CsvParser.parse(file_record),
+         :ok <- SearchKeywords.store_keywords(keyword_list, file_record.id) do
+      FileUploads.completed(file_record)
```

Comment:
Maybe these 2 as a multi. So if the job fails, it does not store keywords twice 💡

Reply:
@rosle Added in [a37c4cb](https://github.com/rafayet-monon/elixir_search_extractor/pull/39/commits/a37c4cbbb26c74fcd6bdd8ab6b7527bf3c0872c2)